src/app.md

```diff
@@ -725,6 +726,50 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-semver-latest]: #ex-semver-latest
+<a name="ex-semver-latest"></a>
+## Find the latest version satisfying given range
+[![semver-badge]][semver] [![cat-config-badge]][cat-config]
+
+Given an array of unparsed version `&str`s, iterate and filter against a semver `VersionReq` and find the latest `Version`.
+
+```rust
+#[macro_use]
+extern crate error_chain;
+extern crate semver;
+
+use semver::Version;
+use semver::VersionReq;
+
+mod errors {
+    error_chain! { }
+}
+
+error_chain! {
+    foreign_links {
+        SemVer(semver::SemVerError);
+    }
+}
+
+fn run() -> Result<()> {
+    let vset = ["0.9.0", "1.0.0", "1.0.1"];
+    let r = VersionReq::parse("<= 1.0.0")
+        .chain_err(|| "unable to parse VersionReq")?;
+
+    let vmax = vset.iter()
+                   .filter_map(|s| Version::parse(s).ok())
+                   .filter(|s| r.matches(&s))
+                   .max()
+                   .chain_err(|| "unable to find max Version")?;
```

Comment:
I'd suggest replacing the `chain_err` with `ok_or_else` to identical effect or just leave it as an `Option` without any unwrapping and test the version_max (suggested variable name) with `Some(Version { ..... })`

Also, I'd recommend to replace the "unable to find max Version" with something along the lines of "Unable to find version satisfying given range".

Reply:
I tried just removing the `chain_err` and leaving it as `.max()?;` but it results in a compiler error `the trait bound `Error: std::convert::From<std::option::NoneError>` is not satisfied`.  So I just replaced the `chain_err` with `ok_or_else` instead.